ci/build_wheel.sh

```diff
@@ -33,9 +33,49 @@ rapids-generate-version > ./VERSION
 
 RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
 
+touch /tmp/constraints-build.txt
+
+if [[ "${package_name}" == "cuspatial" ]]; then
+  # Downloads libcuspatial wheel from this current build,
+  # then ensures 'cuspatial' wheel builds always use the 'libcuspatial' just built in the same CI run.
+  RAPIDS_PY_WHEEL_NAME="libcuspatial_${RAPIDS_PY_CUDA_SUFFIX}" rapids-download-wheels-from-s3 cpp /tmp/libcuspatial_dist
+  echo "libcuspatial-${RAPIDS_PY_CUDA_SUFFIX} @ file://$(echo /tmp/libcuspatial_dist/libcuspatial_*.whl)" > /tmp/constraints-build.txt
+fi
+
+rapids-logger "Generating build requirements"
+declare -r matrix_selectors="cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};cuda_suffixed=true"
+
+rapids-dependency-file-generator \
+  --output requirements \
+  --file-key "py_build_${package_name}" \
+  --matrix "${matrix_selectors}" \
+| tee /tmp/requirements-build.txt
+
+rapids-dependency-file-generator \
+  --output requirements \
+  --file-key "py_rapids_build_${package_name}" \
+  --matrix "${matrix_selectors}" \
+| tee -a /tmp/requirements-build.txt
+
+rapids-logger "Installing build requirements"
+python -m pip install \
+    -v \
+    --prefer-binary \
+    --constraint /tmp/constraints-build.txt \
+    -r /tmp/requirements-build.txt
+
 cd "${package_dir}"
 
-python -m pip wheel . -w dist -vvv --no-deps --disable-pip-version-check
+rapids-logger "Building '${package_name}' wheel"
+python -m pip wheel \
+    -w dist \
+    -v \
+    --no-build-isolation \
+    --no-deps \
+    --disable-pip-version-check \
+    .
+
+sccache --show-adv-stats
```

Comment:
Anticipating a review comment like "is this left over from debugging?"... I've intentionally left this in. I think it's generally useful to print the `sccache` stats in these CI logs, as they could provide a hint when you experience issues like CI timeouts or out-of-memory.

This only adds 27 lines to the logs, like this:

```text
Compile requests                     28
Compile requests executed            28
Cache hits                           28
Cache hits (c [gcc])                  2
Cache hits (c++ [gcc])                2
Cache hits (cuda [nvcc])             24
Cache misses                          0
Cache timeouts                        0
Cache read errors                     0
Forced recaches                       0
Cache write errors                    0
Compilation failures                  0
Cache errors                          0
Non-cacheable compilations            0
Non-cacheable calls                   0
Non-compilation calls                 0
Unsupported compiler calls            0
Average cache write               0.000 s
Average compiler                  0.000 s
Average cache read hit            0.077 s
Failed distributed compilations       0
Cache location                  s3, name: rapids-sccache-east, prefix: /
Version (client)                0.7.7
```

Reply:
Yes, this is good IMO. I'm also going to add this kind of thing into the telemetry.

Reply:
I agree, I want us to start printing sccache in more jobs so that we can proactively identify these kinds of issues.

Reply:
I opened https://github.com/rapidsai/build-planning/issues/111.